File: OpenTool.Core/Util/CharUtil.cs
namespace OpenTool.Core.Util
{
    public static class CharUtil
    {
        /// <summary>
        /// 检查字符是否为小写字母，小写字母指a~z
        /// </summary>
        /// <param name="ch"></param>
        /// <returns></returns>
        public static bool IsLower(char ch)
        {
            return char.IsLower(ch);
        }

        /// <summary>
        /// 判断是否为大写字母，大写字母包括A~Z
        /// </summary>
        /// <param name="ch"></param>
        /// <returns></returns>
        public static bool IsUpper(char ch)
        {
            return char.IsUpper(ch);
        }

        /// <summary>
        /// 检查是否为数字字符，数字字符指0~9
        /// </summary>
        /// <param name="ch"></param>
        /// <returns></returns>
        public static bool IsNumber(char ch)
        {
            return char.IsNumber(ch);
        }

        /// <summary>
        /// 是否空白符
        /// </summary>
        /// <param name="c"></param>
        /// <returns></returns>
        public static bool IsBlank(char c)
        {
            return char.IsWhiteSpace(c)
                    || c == '\ufeff'
                    || c == '\u202a'
                    || c == '\u0000'
                    // issue#I5UGSQ，Hangul Filler
                    || c == '\u3164'
                    // Braille Pattern Blank
                    || c == '\u2800'
                    // MONGOLIAN VOWEL SEPARATOR
                    || c == '\u180e';
        }
    }
}

File: opentool-core/src/main/java/cn/luxio/opentool/core/util/CharUtil.java
package cn.luxio.opentool.core.util;

/**
 * 字符工具类
 */
public class CharUtil {

    /**
     * 判断字符是否为空白字符
     *
     * @param c 待判断字符
     * @return 为空白字符返回 {@code true}，否则返回 {@code false}
     */
    public static boolean isBlankChar(final char c) {
        return isBlankChar((int) c);
    }

    /**
     * 判断字符是否为非空白字符
     *
     * @param c 待判断字符
     * @return 为空白字符返回 {@code false}，否则返回 {@code true}
     */
    public static boolean isNotBlankChar(final char c) {
        return !isBlankChar((int) c);
    }

    /**
     * 判断 Unicode 码点是否为空白字符
     * <p>
     * 除 Java 标准空白字符外，也包含 BOM、零宽字符等常见不可见字符
     *
     * @param c 待判断 Unicode 码点
     * @return 为空白字符返回 {@code true}，否则返回 {@code false}
     */
    @SuppressWarnings("UnnecessaryUnicodeEscape")
    public static boolean isBlankChar(final int c) {
        return Character.isWhitespace(c)
                || Character.isSpaceChar(c)
                || c == '\ufeff'
                || c == '\u202a'
                || c == '\u0000'
                // issue#I5UGSQ，Hangul Filler
                || c == '\u3164'
                // Braille Pattern Blank
                || c == '\u2800'
                // Zero Width Non-Joiner, ZWNJ
                || c == '\u200c'
                // MONGOLIAN VOWEL SEPARATOR
                || c == '\u180e';
    }
}
